3032/new what to expect core

jaredcwhite opened the pull request:
## Issue Overview

This PR addresses #3032, #3033, #3034, and https://github.com/housingbayarea/bloom/issues/492

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This updates What to Expect-related text on the Get Started, Terms, and Confirmation form steps of public applications so they account for Lottery, FCFS, and Waitlist types.

**Note:** this also includes a first pass at a Gen2-ification of various Markdown styles (mostly ported over from HBA). Probably more cleanup would be good here, but feels out of scope for this particular issue.

## How Can This Be Tested/Reviewed?

There are several listings of different types you can review here:
https://deploy-preview-3087--dev-bloom.netlify.app/listings

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

emilyjablonski commented on sites/public/pages/applications/review/confirmation.tsx:
```diff
@@ -33,26 +33,32 @@ const ApplicationConfirmation = () => {
-      } else {
-        return t("application.review.confirmation.eligibleApplicants.FCFS")
+  const content = useMemo(() => {
+    if (listing?.reviewOrderType == ListingReviewOrder.firstComeFirstServe) {
```
Thoughts on making these switch statements?

jaredcwhite replied:
Sure, a `switch` statement sounds good.

emilyjablonski commented on ui-components/src/global/markdown.scss:
```diff
@@ -1,53 +1,173 @@
+    }
+  }
+
+  /* Used for a larger lead paragraph: */
```
In the future, let's separate out changes like this into a different PR so that we can make an issue, discuss it in planning, point it, shorten the review scope, and make sure that the point estimate we made for the initial issue in planning stays accurate :)

jaredcwhite replied:
Oh, I forgot to mention https://github.com/housingbayarea/bloom/issues/492 in the PR description. It was something that already come up with a previous review round…I didn't want to just port the Markdown styles over from HBA without at least modernizing it a bit, and I do think we should create another issue for further review of our Markdown styles (part of the audit?)

emilyjablonski commented on ui-components/src/locales/general.json:
```diff
@@ -439,13 +441,21 @@
+  "application.review.terms.textSubmissionDate": "This application must be submitted by %{applicationDueDate}.",
+  "application.review.terms.fcfs.text": "* Applicants are applying to currently vacant apartments on a first come, first serve basis.\n\n* Eligibile applicants will be contacted on a first come first serve basis until vacancies are filled.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide supporting documents.\n\n* All of the informa ...
+  "application.review.terms.lottery.text": "* Applicants are applying to enter a lottery for currently vacant apartments.\n\n* Once the application period closes, eligible applicants will be placed in lottery rank order.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide supporting documents.\n\n* All of the information that you have  ...
+  "application.review.terms.waitlist.text": "* Applicants are applying for an open waitlist and not a currently vacant apartment.\n\n* When vacancies become available, eligible applicants will contacted by the property manager on a first come, first serve basis.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide supporting documents.\ ...
```
There is a typo in the copy here - need to update `will contacted` to `will be contacted`

emilyjablonski commented on ui-components/src/locales/general.json:
```diff
@@ -439,13 +441,21 @@
+  "application.start.whatToExpect.lottery.finePrint": "* Applicants are applying to enter a lottery for currently vacant apartments.\n*  Please be aware that each household member can only appear on one application for each listing.\n*  Applicants will be contacted by the property agent in lottery rank order until vacancies are filled.\n*  All of the information that you have provided will be ver ...
+
+  "application.start.whatToExpect.waitlist.steps": "1. First we'll ask about you and the people you plan to live with.\n2. Then, we'll ask about your income.\n3. Finally, we'll see if you qualify for any affordable housing preferences, if applicable.",
+  "application.start.whatToExpect.waitlist.finePrint": "* Applicants are applying for an open waitlist and not a currently available apartment.\n* Please be aware that each household member can only appear on one application for each listing.\n* When vacancies become available, eligible applicants will contacted by the property manager on a first come, first serve basis.\n* All of the information ...
```
There is a typo in the copy here - need to update `will contacted` to `will be contacted`

emilyjablonski commented on ui-components/src/locales/general.json:
```diff
@@ -379,7 +379,9 @@
   "application.review.confirmation.whatExpectSecondparagraph": "Applicants will be contacted in order until vacancies are filled. Should your application be chosen, be prepared to fill out a more detailed application and provide required supporting documents.",
   "application.review.confirmation.whatExpectTitle": "What to expect next",
-  "application.review.confirmation.whatHappensNext": "### What happens next?\n\n* After all applications are submitted, the property manager will begin processing applications.\n\n* %{reviewOrder}\n\n* If you are contacted for an interview, you will need to fill out a more detailed application and provide supporting documents.",
+  "application.review.confirmation.whatHappensNext.fcfs": "### What happens next?\n\n* Eligibile applicants will be contacted on a first come first serve basis until vacancies are filled.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide supporting documents.",
```
There is a typo in the copy here - need to update `eligibile` to `eligible`

emilyjablonski commented on ui-components/src/locales/general.json:
```diff
@@ -379,7 +379,9 @@
-  "application.review.confirmation.whatHappensNext": "### What happens next?\n\n* After all applications are submitted, the property manager will begin processing applications.\n\n* %{reviewOrder}\n\n* If you are contacted for an interview, you will need to fill out a more detailed application and provide supporting documents.",
+  "application.review.confirmation.whatHappensNext.fcfs": "### What happens next?\n\n* Eligibile applicants will be contacted on a first come first serve basis until vacancies are filled.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide supporting documents.",
+  "application.review.confirmation.whatHappensNext.lottery": "### What happens next?\n\n* Once the application period closes, eligible applicants will be placed in order based on lottery rank order.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide supporting documents.",
+  "application.review.confirmation.whatHappensNext.waitlist": "### What happens next?\n\n* Eligibile applicants will be placed on the waitlist on a first come first serve basis until waitlist spots are filled.\n\n* Housing preferences, if applicable, will affect waitlist order.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide suppor ...
```
There is a typo in the copy here - need to update `eligibile` to `eligible`

emilyjablonski commented on ui-components/src/locales/general.json:
```diff
@@ -439,13 +441,21 @@
-  "application.review.terms.textSubmissionDate": "This application must be submitted by %{applicationDueDate}.<br><br>",
-  "application.review.terms.text": "Applicants will be contacted by the leasing agent in lottery and preference order or waitlist order until vacancies are filled. All of the information that you have provided will be verified and your eligibility confirmed. Your application may be removed from the waitlist if you have made any fraudulent statements and duplicate applications from the same househ ...
+  "application.review.terms.textSubmissionDate": "This application must be submitted by %{applicationDueDate}.",
+  "application.review.terms.fcfs.text": "* Applicants are applying to currently vacant apartments on a first come, first serve basis.\n\n* Eligibile applicants will be contacted on a first come first serve basis until vacancies are filled.\n\n* If you are contacted for an interview, you will be asked to fill out a more detailed application and provide supporting documents.\n\n* All of the informa ...
```
There is a typo in the copy here - need to update `eligibile` to `eligible`

emilyjablonski reviewed (changes requested):
I have a few code comments, and then functionally all looks well except for a lottery listing confirmation page - I'm not seeing the preference string.

<img width="406" alt="Screen Shot 2022-10-05 at 2 13 08 PM" src="https://user-images.githubusercontent.com/65367387/194133149-18ac5e08-f3f8-4c42-9e8c-7500704e2dbb.png">

jaredcwhite commented:
@emilyjablonski I fixed the typos and switched to switch (ha!) statements. Regarding the preferences text, it's weird —[ the document](https://docs.google.com/spreadsheets/d/1zrZb9WFRJ2qFPIyfkFGK_BkeQ1_ejk7j9qYuxO35mOs/edit#gid=1891467199) seems to have blacked out those for the Confirmation screen, but not for the confirmation email:

<img width="1043" alt="image" src="https://user-images.githubusercontent.com/658496/194672623-c70b1996-622b-4813-ab6e-d0285b89bdee.png">

I wonder if that was intentional or not?

ludtkemorgan commented on sites/public/pages/applications/review/confirmation.tsx:
```diff
@@ -33,25 +33,34 @@ const ApplicationConfirmation = () => {
+          text: t("application.review.confirmation.whatHappensNext.fcfs"),
+        }
+      case ListingReviewOrder.lottery:
+        lotteryEvent = getLotteryEvent(listing)
```
I could be missing it, especially since I don't know what the different colors in the google doc represent. But I don't see any of the text say they still want the lottery date listed.

jaredcwhite commented:
Changes made. Last round!

emilyjablonski reviewed (approved):
🥳 🥳 🥳
